hudi-spark2/src/main/scala/org/apache/hudi/DataSourceOptionsForSpark2.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi
+
+import org.apache.hudi.common.model.HoodieTableType
+import org.apache.hudi.common.model.OverwriteWithLatestAvroPayload
+
+/**
+ * Options supported for writing hoodie tables.
+ * TODO: This file is partially copied from org.apache.hudi.DataSourceWriteOptions.
+ * Should be removed if Spark 2.x support is dropped.
+ */
+object DataSourceWriteOptionsForSpark2 {
```

Comment:
I'm not sure this is best to have copies `DataSourceWriteOptions` and `DataSourceUtils`. I see in `hudi-client` there is a a module `hudi-spark-client`. Could we refactor these files to go there or in some new `hudi-spark-common`

Reply:
Yea this will avoid us hard copy ```DataSourceWriteOptions``` and ```DataSourceUtils```. My concern is Spark Datasource related files ideally should under ```hudi-spark```. So not sure which is the best practice here..

Maybe create a new module like ```hudi-spark-common``` under ```hudi-spark```?

Reply:
Yeah I don't really think this is manageable, specially as new properties get added it will be a pain to make sure spark2 and spark3 code is in sync. `hudi-spark-common` module under `hudi-spark` seems like the way to me. But lets get vinoth's suggestion on this as well before you implement cc @vinothchandar

Reply:
yes `hudi-spark-common` is the right approach IMO as well.